tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1268,3 +1294,137 @@ def _retrieve_clustermgtd_heartbeat_file(remote_command_executor, clustermgtd_co
+    """Enable protected mode by removing lines related to protected mode in the config, so it will be set to default."""
+    remote_command_executor.run_remote_command(
+        f"sudo sed -i '/'insufficient_capacity_timeout'/d' {clustermgtd_conf_path}"
+    )
+
+
+def _test_disable_fast_capacity_failover(scheduler_commands, remote_command_executor, clustermgtd_conf_path):
+    """Test fast capacity failover has no effect on cluster when it is disabled."""
+    # Disable fast capacity failover by setting insufficient_capacity_timeout < 0
+    _set_insufficient_capacity_timeout(remote_command_executor, 0, clustermgtd_conf_path)
+    # submit a job to trigger insufficient capacity
+    job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={"command": "sleep 30", "nodes": 2}
+    )
+    # wait till the node failed to launch
+    retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))(assert_errors_in_logs)(
+        remote_command_executor,
+        ["/var/log/parallelcluster/slurm_resume.log"],
+        [
+            "InsufficientInstanceCapacity",
+        ],
+    )
+    # assert that ice node is detected as unhealthy node
+    retry(wait_fixed=seconds(20), stop_max_delay=minutes(2))(assert_errors_in_logs)(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        [
+            "Found the following unhealthy dynamic nodes",
+        ],
+    )
+    # Assert that clustermgtd log doesn't contains insufficient capacity compute resources
+    assert_no_msg_in_logs(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        ["The following compute resources are in down state due to insufficient capacity"],
+    )
+    return job_id
+
+
+def assert_job_requeue_in_time(scheduler_commands, job_id):
+    """Test that job will requeue to a different compute resource after AuthInfo=cred_expire=70 timeout."""
+    scheduler_commands.wait_job_completed(job_id)
+    scheduler_commands.assert_job_succeeded(job_id)
+    submit_time = datetime.strptime(scheduler_commands.get_job_submit_time(job_id), "%Y-%m-%dT%H:%M:%S")
+    eligible_time = datetime.strptime(scheduler_commands.get_job_eligible_time(job_id), "%Y-%m-%dT%H:%M:%S")
+    wait_seconds = (eligible_time - submit_time).total_seconds()
+    # Test it takes less than 2 minutes a job is re-queued to a different compute resource."""
+    assert_that(wait_seconds).is_less_than_or_equal_to(120)
+
+
+def _test_enable_fast_capacity_failover(scheduler_commands, remote_command_executor, job_id, clustermgtd_conf_path):
+    _enable_fast_capacity_failover(remote_command_executor, clustermgtd_conf_path)
+    # reduce insufficient_capacity_timeout to quicker reset compute resources
+    _set_insufficient_capacity_timeout(remote_command_executor, 120, clustermgtd_conf_path)
+    # trigger insufficient capacity
+    retry(wait_fixed=seconds(20), stop_max_delay=minutes(7))(assert_errors_in_logs)(
```

Comment:
nit: probably you need much less than 7 mins

Reply:
Reduce to 3 minutes